cmake/Dependencies.cmake

```diff
@@ -4,7 +4,19 @@ set(Caffe_LINKER_LIBS "")
 # ---[ Boost
 find_package(Boost 1.46 REQUIRED COMPONENTS system thread filesystem)
 include_directories(SYSTEM ${Boost_INCLUDE_DIR})
-list(APPEND Caffe_LINKER_LIBS ${Boost_LIBRARIES})
+link_directories(
+    ${Boost_LIBRARY_DIR_RELEASE}
+)
+
+if(NOT DEFINED MSVC OR WITH_PREBUILD_MSVC_DEPENDENCIES )
```

Comment:
I think we should avoid adding the link directories as this is considered bad practice in CMake. We should append Boost libraries to the list and add the BOOST_ALL_NO_LIB define.